operation/fhir-operation-reindex/src/main/resources/retrieve-index.json

```diff
@@ -0,0 +1,53 @@
+{
+	"resourceType": "OperationDefinition",
+	"id": "retrieve-index",
+	"text": {
+		"status": "generated",
+		"div": "<div xmlns='http://www.w3.org/1999/xhtml'><h2>retrieve-index</h2><p>OPERATION: Retrieve index IDs of resources available to reindex</p><p>The retrieve-index operation retrieves the index IDs from the database. These index IDs can then be passed into the reindex operation to indicate the specific resources to reindex.</p><p>URL: [base]/$retrieve-index</p></div>"
+	  },
+	"url": "http://ibm.com/fhir/OperationDefinition/retrieve-index",
+	"name": "RetrieveIndex",
+	"status": "draft",
+	"kind": "operation",
+	"publisher": "IBM FHIR Server",
+	"date": "2021-06-11",
+	"description": "The retrieve-index operation retrieves the index IDs from the database.",
+	"code": "retrieve-index",
+	"system": true,
+	"type": false,
```

Comment:
If we support this one at the type level, that would allow us to add selective reindexing to the fhir-bucket client:  https://github.com/IBM/FHIR/issues/2525

In #2489 we did the same thing for the `$reindex` operation (i.e. made it work on the type level in addition to the system level).

Reply:
Yeah, that makes sense.  I can add `$retrieve-index` support at the type level.

Reply:
Added.